### PR TITLE
Fixed Media Sync file not found error

### PR DESCRIPTION
**Pull request by tomac120**
Worked around the temp file not found error by using ZipInputStream
(does not require the file to be written to the disk). In any case, the
data was already in memory from the HTTP input stream, so no additional
device memory should be used than before.

### CODE REVIEW

**Comment by timrae** [comment]
@tomac120 
Also, could you please remove some of the stray comments, and squash into one commit?

@hssm what do you think? 

There seems to currently be 74 users who have suffered this problem in the last couple of weeks, some of whom experience the problem repeatedly on one day, but very few who have continually experienced it on multiple different days... so I guess the root cause is a problem with the device accessing the SD card... in which case this change makes sense

```
02-16 07:17:41.144 E/AnkiDroid( 9987): MediaSyncer/ Failed to create temp media sync zip file
02-16 07:17:41.144 E/AnkiDroid( 9987): java.io.FileNotFoundException: /storage/emulated/0/AnkiDroid/tmpSyncFromServer.zip: open failed: ENOENT (No such file or directory)
02-16 07:17:41.144 E/AnkiDroid( 9987): at libcore.io.IoBridge.open(IoBridge.java:416)
02-16 07:17:41.144 E/AnkiDroid( 9987): at java.io.RandomAccessFile.<init>(RandomAccessFile.java:118)
02-16 07:17:41.144 E/AnkiDroid( 9987): at java.io.RandomAccessFile.<init>(RandomAccessFile.java:150)
02-16 07:17:41.144 E/AnkiDroid( 9987): at java.util.zip.ZipFile.<init>(ZipFile.java:146)
02-16 07:17:41.144 E/AnkiDroid( 9987): at com.ichi2.libanki.sync.RemoteMediaServer.downloadFiles(RemoteMediaServer.java:109)
```

it does differ quite a bit from the python though, so I guess we need to at least add some more comments about the background of the issue to the code itself

**Comment by tomac120** [comment]
I've removed that one commit and some commented lines.

**Comment by timrae** [comment]
Please squash to one commit

**Comment by tomac120** [comment]
How is that done?

**Comment by timrae** [comment]
http://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git

File: AnkiDroid/src/main/java/com/ichi2/libanki/Media.java [hunk 1]
```diff
@@ -874,32 +875,64 @@ public void forceResync() {
      * caller to save the zip to disk first. This also spares us from holding the entire downloaded data in memory,
      * which is not practical on a mobile device.
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Media.java]: This javadoc will need to be updated as well

**Comment by hssm** [comment]
I'll review it closely and test it tomorrow. It's fine if it's not perfect, I can fix things up and add comments myself after it's merged.

In general I think this is a good idea. There should be a TODO comment somewhere about trying this out because I suspect it'll be significantly faster than the current approach (I might even benchmark it).

Btw the maximum file size allowed by AnkiWeb is 100MB, so we have to consider that with this change as well, but realistically I don't think it'll be a problem (at the very least it will be less of a problem than the error it will fix).

Thanks a lot for the contribution!

**Comment by tomac120** [comment]
https://github.com/ankidroid/Anki-Android/pull/740

**Comment by tomac120** [comment]
Technically, from the old code the request was already in memory and then written to the disk/SD card, so nothing new in that respect. I think the 100 MB file upper limit is somewhat excessive until video is added.

**Comment by tomac120** [comment]
A 2 MB audio file is certainly more than 1 minute long (not suitable on the card, so its not surprising this issue hasn't come up)

**Pull request closed without merging**